analytics/js_tracking_api.rst

```diff
@@ -0,0 +1,859 @@
+| movie | msi | msp | odb | odf  | odg | odp | ods | odt  | ogg | ogv  | pdf | phps    | png | ppt |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| qt    | qtm | ra  | ram | rar  | rpm | sea | sit | tar  | tbz | tbz2 | tgz | torrent | txt | wav |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+| wma   | wmv | wpd | xls | xml  | z   | zip |     |      |     |      |     |         |     |     |
++-------+-----+-----+-----+------+-----+-----+-----+------+-----+------+-----+---------+-----+-----+
+
+You can add extension to this list using ``addDownloadExtensions`` function::
+
+    _paq.push(["addDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"mhj|docx"``
+
+
+Example of usage::
+
+    _paq.push(["addDownloadExtensions", "mhj|docx"]);
+
+This list can be rewrote using ``setDownloadExtensions`` function::
+
+    _paq.push(["setDownloadExtensions", extensions]);
+
+.. data:: extensions
+
+    **Required** String containing extensions separated by ``|`` for example ``"7z|apk|mp4"``
+
+
+Example of usage::
+
+    _paq.push(["setDownloadExtensions", "7z|apk|mp4"]);
+
+To track clicking a link as an download using css class simply add ``piwik_download`` class to link element.
+
+This class name can be changed, use ``setDownloadClasses`` to define which CSS class should be tracked::
+
+    _paq.push(["setDownloadClasses", className]);
+
+
+.. data:: className
+
+    **Required** String with css class that should be tracked instead of ``piwik_download``
+
+Example of usage::
+
+    _paq(["setDownloadClasses", "track-this-link-for-download"]);
+
+
+If you want to use JS instead you can add ``trackLink`` function to element ``onClick`` attribute::
+
+    _paq.push(["trackLink", linkAddress, "download", dimensions]);
+
+.. data:: linkAddress
+
+    **Required** Address that link points to.
+
+.. data:: dimensions
+
+    **Optional**  :ref:`Custom dimension<Custom Dimensions>` that should be tracked with this action. Can be multiple dimensions.
+    Written as object property using ``dimension{ID}`` notation.
+
+    Example::
+
+        {
+           dimension1: "example value",
+           dimension2: "example value"
+        }
+
+Example of usage
+
+.. code-block:: html
+
+    <button onClick = function(){_paq.push(["trackLink", "http://www.example.com/example.xrt", "download"]);}>
+        Click me!
+    </button>
+
+
+After each outbound link there is small time frame after which the file will download that will
+ensure there is enough time to track download.
+That time frame is set to 500ms by default. To modify it you can use ``setLinkTrackingTimer`` function::
+
+    _paq.push(["setLinkTrackingTimer" time]);
+
+.. data:: time
+
+    **Required** Time in ms between user interaction and downloading file.
+
+
+Disabling tracking
```

Comment:
I would like to have it near CSS enable section, but that is separated between link and download sections. Can they be moved to one place (outside link and downloads sections)?

It looks like this function is reverse to 2 other specialized functions for links and downloads. I'm not sure what to do with it. In the future it may be better to have one function and one reverse function to keep it clean.

Reply:
These things are written like that in the tracker object reference